scripts/src/javatests/org/oppia/android/scripts/coverage/CoverageReporterTest.kt

```diff
@@ -530,11 +630,55 @@ class CoverageReporterTest {
     CoverageReporter(
       tempFolder.root.absolutePath,
       coverageReportContainer,
-      ReportFormat.HTML
+      ReportFormat.HTML,
+      testExemptions
     ).generateRichTextReport()
 
     assertThat(outContent.toString().trim()).contains(
       "The file $exemptedFilePath is exempted from coverage analysis"
     )
   }
+
+  private fun readFinalMdReport(): String {
+    return File(
+      "${tempFolder.root}" +
+        "$coverageDir/CoverageReport.md"
+    ).readText()
+  }
+
+  private fun createTestFileExemptionTextProto():
+    Map<String, TestFileExemptions.TestFileExemption> {
+      val testFileExemptions = TestFileExemptions.newBuilder()
+        .addTestFileExemption(
+          TestFileExemption.newBuilder()
+            .setExemptedFilePath("TestExempted.kt")
+            .setTestFileNotRequired(true)
+            .build()
+        )
+        .addTestFileExemption(
+          TestFileExemption.newBuilder()
+            .setExemptedFilePath("coverage/main/java/com/example/HighCoverageExempted.kt")
+            .setOverrideMinCoveragePercentRequired(101)
+            .build()
+        )
+        .addTestFileExemption(
+          TestFileExemption.newBuilder()
+            .setExemptedFilePath("coverage/main/java/com/example/LowCoverageExempted.kt")
+            .setOverrideMinCoveragePercentRequired(0)
+            .build()
+        )
+        .build()
+
+      val testExemptionPb = "test_exemption.pb"
+      val coverageTestExemptionTextProto = tempFolder.newFile(testExemptionPb)
+      coverageTestExemptionTextProto.outputStream().use { outputStream ->
+        testFileExemptions.writeTo(outputStream)
+      }
+
+      val testFileExemptionsFromFile =
+        TestFileExemptions.parseFrom(coverageTestExemptionTextProto.inputStream())
```

Comment:
Not much value in parsing it--I suggest just using the exact structure that was assembled here, instead.

Reply:
Yeah, got it. Removed the parsing step.